spec/isec2_linux.go

```diff
@@ -51,3 +55,26 @@ func isEC2() bool {
 
 	return false
 }
+
+func isEC2UUID(uuid string) bool {
+	conds := func(uuid string) bool {
+		if strings.HasPrefix(uuid, "ec2") || strings.HasPrefix(uuid, "EC2") {
+			return true
+		}
+		return false
+	}
+
+	if conds(uuid) {
+		return true
+	}
+
+	// Check as littele endian.
+	// see. https://docs.aws.amazon.com/ja_jp/AWSEC2/latest/UserGuide/identify_ec2_instances.html
+	fields := strings.Split(uuid, "-")
+	decoded, _ := hex.DecodeString(fields[0]) // fields[0]: UUID time_low(uint32)
+	r := bytes.NewReader(decoded)
+	var data uint32
+	binary.Read(r, binary.LittleEndian, &data)
+
+	return conds(fmt.Sprintf("%x", data))
```

Comment:
uuid string possibly return 0xEC, 0x50 ?

https://play.golang.org/p/zQa6ka44eUp

Reply:
Sorry, I understand what you want to fix. https://docs.aws.amazon.com/ja_jp/AWSEC2/latest/UserGuide/identify_ec2_instances.html